### DISCUSSION TITLE
forme_set hidden inputs rendered outside form

### DISCUSSION THREAD

**Discussion by teamschlaflos** [jeremyevans/forme#122] [General]
I'm having an issue where the two hidden forme_set plugin fields are appended to the DOM tree outside the form element. 

When submitting the form I get the following error:

```
Roda::RodaPlugins::FormeSet::Error:
  _forme_set_data parameter not submitted, make sure the forme_set Sequel plugin is loaded
```

In the screenshot the `<form>` is closed, a bit further down there are the two hidden input fields (_forme_set_data & _forme_set_data_hmac):

![image](https://github.com/jeremyevans/forme/assets/157237738/6237bc19-6bee-4f17-901c-9bba0329f037)

Any ideas what influences the placement of the elements?

**Comment by jeremyevans** [comment]
That looks like a bug.  Can you post a self contained example or failing spec that shows that behavior? The _forme_set_data and _forme_set_data_hmac fields should be emitted into the template when the related block is exited, before the closing form tag, and I'm not sure why that isn't happening in this case.

**Comment by teamschlaflos** [comment]
The issue is related to an unbalanced DOM tree, when removing an excessive `</div>` element the hidden form fields will be added correctly within the form tag. Thanks!

**Reply by jeremyevans** [reply]
Ah, so the tags were inside the form tag in the HTML, but because the HTML was invalid, when the browser built the page, it rearranged the DOM to place the tags after the form tag?

**Reply by teamschlaflos** [reply]
Yes! I double checked and seems you are right! The rendered response is always correct (hidden inputs within the form), but when the browser interprets the bad html, in Developer Tools (_Elements_) it looks like the two fields are rendered outside the form tag.
